persisted.go

```diff
@@ -7,11 +7,19 @@ package mirbft
 
 import (
 	"fmt"
+	"io"
 	"sort"
 
 	pb "github.com/IBM/mirbft/mirbftpb"
+	"github.com/pkg/errors"
 )
 
+//go:generate counterfeiter -o mock/storage.go -fake-name Storage . Storage
+
+type Storage interface {
+	Load(index uint64) (*pb.Persisted, error)
```

Comment:
What is 'index' here?

Reply:
Discussed offline -- this makes sense now.  Index will point out a persisted log entry, and be supplied by the lib consumer at startup.  This will allow the consumer to reprocess from an arbitrary point, even beyond the normal watermarks.